src/contentmanager.cpp

```diff
@@ -189,6 +188,22 @@ QString ContentManager::downloadBook(const QString &id)
     return QString::fromStdString(download->getDid());
 }
 
+void ContentManager::eraseBook(const QString& id)
+{
+    kiwix::Book book = mp_library->getBookById(id);
+    QString dirName = QString::fromUtf8(getDataDirectory().c_str());
+    QString fileSelection = QString::fromUtf8(book.getPath().c_str()).remove(0, getDataDirectory().size() + 1) + "*";
+    QDir dir(dirName, { fileSelection });
+    for(const QString& filename: dir.entryList()) {
+        dir.remove(filename);
+        qInfo() << "fichier "<<filename;
+    }
+    mp_library->removeBookFromLibraryById(id);
+    mp_library->save();
+    emit(mp_library->booksChanged());
+    emit(booksChanged());
```

Comment:
There is already a slot connect to `mp_library->booksChanged()` that emit `bookChanged()`. So no need to emit twice.